packages/backend/src/services/CatalogService/CatalogService.ts

```diff
@@ -524,17 +542,21 @@ export class CatalogService<
                 userUuid: user.userUuid,
             });
 
-        const catalogMetrics = await this.searchCatalog(
+        // ! Pagination here might not work as expected because there's some filtering being applied after the data is fetched
+        const paginatedCatalogMetrics = await this.searchCatalog(
             projectUuid,
             userAttributes,
             search,
             CatalogType.Field,
             CatalogFilter.Metrics,
+            paginateArgs,
         );
```

Comment:
This is important, we paginate on the query, but since we then apply some filters on the JS side (e.g. to check user attributes) the page might return less results than the `pageSize`
Not sure how we can solve this yet, but it is also a problem in other places of the app